cdk/cdk/cdk_stack.py

```diff
@@ -69,44 +88,75 @@ def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
             ),
         )
 
-        ## State Machine Steps
-        start_state = stepfunctions.Pass(self, "StartState")
-        discover_task = tasks.LambdaInvoke(
-            self, "Discover Granules Task", lambda_function=discover_lambda
+        ## CMR Workflow State Machine Steps
+        cmr_start_state = stepfunctions.Pass(self, "CMR Discovery StartState")
+        s3_start_state = stepfunctions.Pass(self, "S3 Discovery StartState")
+        cmr_discover_task = tasks.LambdaInvoke(
+            self, "CMR Discover Granules Task", lambda_function=cmr_discover_lambda
         )
+        s3_discover_task = tasks.LambdaInvoke(
+            self, "S3 Discover COGs Task", lambda_function=s3_discover_lambda
+        )
+
         generate_cog_task = tasks.LambdaInvoke(
             self, "Generate COG Task", lambda_function=generate_cog_lambda
         )
-        generate_stac_item_task = tasks.LambdaInvoke(
+        cmr_generate_stac_item_task = tasks.LambdaInvoke(
             self,
-            "Generate STAC Item Task",
+            "CMR Generate STAC Item Task",
             lambda_function=generate_stac_item_lambda,
             input_path="$.Payload",
         )
 
+        s3_generate_stac_item_task = tasks.LambdaInvoke(
+            self,
+            "S3 Generate STAC Item Task",
+            lambda_function=generate_stac_item_lambda
+        )
+
+
         map_cogs = stepfunctions.Map(
             self,
-            "Map State",
+            "Map COG and STAC Item Generator",
+            max_concurrency=10,
+            items_path=stepfunctions.JsonPath.string_at("$.Payload"),
+        )
+
+        # Generate a cog and create stac item for each element
+        map_cogs.iterator(generate_cog_task.next(cmr_generate_stac_item_task))
+
+
+        map_stac_items = stepfunctions.Map(
+            self,
+            "Map STAC Item Generator",
             max_concurrency=10,
             items_path=stepfunctions.JsonPath.string_at("$.Payload"),
         )
 
         # Generate a cog and create stac item for each element
-        map_cogs.iterator(generate_cog_task.next(generate_stac_item_task))
+        map_stac_items.iterator(s3_generate_stac_item_task)
+
+
+        cmr_wflow_definition = cmr_start_state.next(cmr_discover_task).next(map_cogs)
+
+        cmr_wflow_state_machine= stepfunctions.StateMachine(
+            self, f"{collection}-COG-StateMachine", definition=cmr_wflow_definition
+        )
 
-        definition = start_state.next(discover_task).next(map_cogs)
+        s3_wflow_definition= s3_start_state.next(s3_discover_task).next(map_stac_items)
 
-        simple_state_machine = stepfunctions.StateMachine(
-            self, f"{collection}-COG-StateMachine", definition=definition
+        s3_wflow_state_machine = stepfunctions.StateMachine(
+            self, f"{bucket}-{prefix}-COG-StateMachine", definition=s3_wflow_definition
         )
 
+
         # Rule to run it
         rule = events.Rule(
```

Comment:
In general, I think these schedule rules will be per state machine. However in this particular case I don't think we are expecting new files on a regular cadence so I don't there there should be any rule for the discover S3 -> generate STAC workflow we are creating.